### ISSUE TITLE
Feature Idea: linux-app should use ~/.config for anything that needs write access

### ISSUE THREAD

**Issue by GeopJr** [mattermost/focalboard#1410]
## Summary

linux-app should use `$XDG_CONFIG_HOME` and `$HOME/.config` to save everything that requires write access.

## How important this is to me and why

Importance: High

Use cases:
1. Distribute as a package
2. Per-user config and database

## Additional context/similar features

I've spent some time working on packaging focalboard into the following formats: rpm, deb, flatpak
While everything is going as planned there's a major issue; apps don't have write access to their installed location by default (in focalboard's case, `/opt/focalboard-app`).

While there might be workarounds, the best practice would be to move anything that requires write access to `.config` (`dbconfig` & `filespath` (?)). Doing so will also allow each user to have their own .db as well as ease backups and restores.

From a look at the source, it seems that `config.json` is being ignored (for the app at least).

Here's some solutions I came up with:

- Use the values from the config.json in combination with the hardcoded ones.
- Use cli flags or env vars to override them
- Change the hardcoded value to use `~/.config`
```go
// linux/main.go
...
func runServer(port int) (*server.Server, error) {
	logger, _ := mlog.NewLogger()
+	configFolder := os.Getenv("XDG_CONFIG_HOME")
+	if len(configFolder) == 0 {
+		configFolder = os.Getenv("HOME") + "/.config"
+	}

+	configFolder = configFolder + "/focalboard-app"

+	if _, err := os.Stat(configFolder); os.IsNotExist(err) {
+		_ = os.Mkdir(configFolder, 0755)
+	}

	config := &config.Configuration{
		ServerRoot:              fmt.Sprintf("http://localhost:%d", port),
		Port:                    port,
		DBType:                  "sqlite3",
+		DBConfigString:          configFolder + "/focalboard.db",
		UseSSL:                  false,
		SecureCookie:            true,
		WebPath:                 "./pack",
		FilesDriver:             "local",
+		FilesPath:               configFolder + "/focalboard_files",
		Telemetry:               true,
		WebhookUpdate:           []string{},
		SessionExpireTime:       259200000000,
		SessionRefreshTime:      18000,
		LocalOnly:               false,
		EnableLocalMode:         false,
		LocalModeSocketLocation: "",
		AuthMode:                "native",
	}
...
```

Some notes on them:
- The third one could be used with #1018 which would handle the creation of the config folder
- #1256 (for linux-app) would also be useful for per-user configuration (plus wouldn't require root to edit it)
- The third solution would unfortunately break existing installations as the hardcoded path would change
- Both the first and second solutions are easy to use with the packages

After a solution to this is found, I'll be able to continue working on packaging focalboard in more formats (rpm, deb, flatpak, appimage, snap).

**Comment by CodeCubeNeo** [comment]
I strongly agree, any updates on this? My main point also being to distribute this app as a flatpak!

**Comment by skewballfox** [comment]
while I mostly agree, `$XDG_CONFIG_HOME` (i.e. `~/.config`) is for configuration files whereas `$XDG_DATA_HOME` (i.e. `~/.local/share`) is where applications store state. 

Though there are applications that ignore this and use config home for both, (vscode and obsidian come to mind). I'd definitely appreciate following spec though.
